subsys/net/ip/tcp2.c

```diff
@@ -1083,46 +1091,61 @@ static struct tcp *tcp_conn_alloc(void)
 
 	ret = k_mem_slab_alloc(&tcp_conns_slab, (void **)&conn, K_NO_WAIT);
 	if (ret) {
+		NET_ERR("Cannot allocate slab");
 		goto out;
 	}
 
 	memset(conn, 0, sizeof(*conn));
 
+	if (CONFIG_NET_TCP_RECV_QUEUE_TIMEOUT) {
+		conn->queue_recv_data = tcp_rx_pkt_alloc(conn, 0);
+		if (conn->queue_recv_data == NULL) {
+			NET_ERR("Cannot allocate %s queue for conn %p", "recv",
+				conn);
+			goto fail;
+		}
+	}
+
+	conn->send_data = tcp_pkt_alloc(conn, 0);
+	if (conn->send_data == NULL) {
+		if (CONFIG_NET_TCP_RECV_QUEUE_TIMEOUT) {
+			tcp_pkt_unref(conn->queue_recv_data);
```

Comment:
Don't you think it's more reliable/easy to follow pattern when "cleanup actions" are done altogether in "exception handler" (i.e. under the "fail" label). That would require an extra `if (conn->queue_recv_data)`, but would improve code clarity and maintainability IMHO.

Reply:
Sure, that can be done.